CMakeLists.txt

```diff
@@ -692,6 +692,11 @@ if(HPX_WITH_PARCEL_COALESCING)
   hpx_add_config_define(HPX_HAVE_PARCEL_COALESCING)
 endif()
 
+# Developer tools
+hpx_option(HPX_WITH_VIM_YCM STRING
+  "Generate HPX completion file for VIM YouCompleteMe plugin and save in specified location"
+  "" ADVANCED)
```

Comment:
Isn't the YCM file provided by us? Why does the user need to pass a string here?
I would rather prefer to have a boolean option and addition YCM_FILE option. This would make those options consistent